### DISCUSSION TITLE
Problem with testing firedrake

### DISCUSSION THREAD

**Discussion by AlcaeusX** [firedrakeproject/firedrake#2798] [Firedrake support]
Hello! I am new to firedrake, and I am facing some problems after installing it. I use a WSL Ubuntu 18.04 and work on VSC. I believe I correctly installed firedrake, following all the steps on the site. Used the python3 firedrake-install command. 

This message appeared while installing PETSC: "The git repository at "/home/alkaios/firedrake/src/petsc" has too many active changes, only a subset of Git features will be enabled."

The installation was successful, but when I tried to test firedrake (cd $VIRTUAL_ENV/src/firedrake
pytest tests/regression/ -k "poisson_strong or stokes_mini or dg_advection") I got the following errors:

""
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/_pytest/config.py", line 365, in _importconftest
    mod = conftestpath.pyimport()
  File "/usr/lib/python2.7/dist-packages/py/_path/local.py", line 668, in pyimport
    __import__(modname)
  File "/usr/lib/python2.7/dist-packages/_pytest/assertion/rewrite.py", line 213, in load_module
    py.builtin.exec_(co, mod.__dict__)
  File "/usr/lib/python2.7/dist-packages/py/_builtin.py", line 221, in exec_
    exec2(obj, globals, locals)
  File "<string>", line 7, in exec2
  File "/home/alkaios/firedrake/src/firedrake/tests/conftest.py", line 6, in <module>
    from pyadjoint.tape import get_working_tape
ImportError: No module named pyadjoint.tape
ERROR: could not load /home/alkaios/firedrake/src/firedrake/tests/conftest.py ""

I also tried to "import firedrake" in python and python3 in the venv, and it didn't raise any errors, but anything I tried after that was marked as "not defined" 
( >>> import firedrake
>>> fe = FiniteElement("RTCE", quadrilateral, 2, variant="equispaced")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'FiniteElement' is not defined)

Any help would be greatly appreciated!!

**Comment by dham** [comment]
The traceback indicates that you are using `pytest` from a system Python 2.7 install rather that one installed in the venv. Make sure the venv is active and then use `python -m pytest` rather than just `pytest`. This is more likely to find the correct Pytest.

The last issue is simpy how Python import works. If you want to use Firedrake names on their own (i.e. `FiniteElement` and not `firedrake.FiniteElement` then you need `from firedrake import *`.

**Reply by AlcaeusX** [reply]
Dear dham, thank you for your reply. I followed the steps and still get the same problem 

(firedrake) test@test:~/firedrake/src/firedrake$ sudo python -m pytest -k "poisson_strong or stokes_mini or dg_advection"
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/_pytest/config.py", line 365, in _importconftest
    mod = conftestpath.pyimport()
  File "/usr/lib/python2.7/dist-packages/py/_path/local.py", line 668, in pyimport
    __import__(modname)
  File "/usr/lib/python2.7/dist-packages/_pytest/assertion/rewrite.py", line 213, in load_module
    py.builtin.exec_(co, mod.__dict__)
  File "/usr/lib/python2.7/dist-packages/py/_builtin.py", line 221, in exec_
    exec2(obj, globals, locals)
  File "<string>", line 7, in exec2
  File "/home/alkaios/firedrake/src/firedrake/tests/conftest.py", line 6, in <module>
    from pyadjoint.tape import get_working_tape
ImportError: No module named pyadjoint.tape
ERROR: could not load /home/alkaios/firedrake/src/firedrake/tests/conftest.py

**Reply by dham** [reply]
Don't use sudo. You really do not want to be running things as root.

Run `python -c "import sys; print(sys.executable)"` and verify that you are actually running the Python from your venv. If you are, then you try `python -m pytest ...`. If you are not then your issue is that your environment is somehow mucked up. Try launching a new terminal and activating the venv there.

**Reply by AlcaeusX** [reply]
Ok, thank you! I followed the steps as follows: 

(firedrake) test@test:~$ python -c "import sys; print(sys.executable)"
/home/alkaios/firedrake/bin/python

Now i get this error:

(firedrake) test@test:~/firedrake/src/firedrake$ python -m pytest -k "poisson_strong or stokes_mini or dg_advection"
ImportError while loading conftest '/home/alkaios/firedrake/src/firedrake/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from pyadjoint.tape import get_working_tape
E   ModuleNotFoundError: No module named 'pyadjoint.tape'

**Reply by dham** [reply]
That looks like your install did not complete successfully. Please post `firedrake-install.log` which you will find in your home folder.

**Reply by AlcaeusX** [reply]
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/10899772/firedrake-install.log)

**Reply by AlcaeusX** [reply]
I noticed there are problems with the installation so I removed everything and installed everything back from WSL2 to firedrake. Now I get this error when testing:

ImportError while loading conftest '/home/firedrake/firedrake/src/firedrake/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from pyadjoint.tape import get_working_tape
E   ModuleNotFoundError: No module named 'pyadjoint'

**Reply by dham** [reply]
Your previous install failed during petsc configuration. It looks like this one may have too. Can you please post `firedrake/src/petsc/configure.log`

**Reply by AlcaeusX** [reply]
the configure.log appears to be corrupted and the system won't let me upload it or open it. Please find attached the install.log
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/10910981/firedrake-install.log)

**Reply by dham** [reply]
That just tells us the same thing: failure occurred while configuring PETSc. Can you zip the config file and upload that?

**Reply by AlcaeusX** [reply]
[petsc.zip](https://github.com/firedrakeproject/firedrake/files/10911050/petsc.zip)
Hopefully this will work. 
Thank you!

**Reply by dham** [reply]
That's `configure`. I'm afraid we need `configure.log`

**Reply by AlcaeusX** [reply]
unfortunately when I try to zip it it appears: 
! \\wsl.localhost\Ubuntu\home\firedrake\firedrake\src\petsc\configure.zip: Cannot open \\wsl.localhost\Ubuntu\home\firedrake\firedrake\src\petsc\configure.log
 The system cannot resolve the file name.

**Reply by AlcaeusX** [reply]
I erased and reinstalled everything. Same errors, but now the configure.log is not corrupted, so I am attaching it. Hope it helps!!
[configure.log](https://github.com/firedrakeproject/firedrake/files/10923609/configure.log)

**Reply by connorjward** [reply]
You are having trouble installing NetCDF so my guess is you're hitting [this issue](https://github.com/firedrakeproject/firedrake/issues/2803). We've updated `firedrake-install` so fingers crossed things will work if you fetch an updated install script.

**Reply by AlcaeusX** [reply]
I clean installed again, and it is working fine now! Thank you for updating the script!
